runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -287,5 +316,78 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	return spine;
 }
 
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+#if !defined(LINUX) || !defined(J9VM_ENV_DATA64) || defined(OMRZTPF)
+/* Double map is only supported on LINUX 64 bit Systems for now */
+#error "Platform not supported by Double Map API"
+#endif /* !definedLINUX || !defined(J9VM_ENV_DATA64) || defined(OMRZTPF) */
+void * 
+MM_IndexableObjectAllocationModel::doubleMapArraylets(MM_EnvironmentBase *env, J9Object *objectPtr) 
+{
+	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(env);
+	J9JavaVM *javaVM = extensions->getJavaVM();
+	PORT_ACCESS_FROM_ENVIRONMENT(env);
+
+	GC_ArrayletLeafIterator arrayletLeafIterator(javaVM, (J9IndexableObject *)objectPtr);
+	UDATA arrayletLeafCount = arrayletLeafIterator.getNumLeafs();
+	MM_Heap *heap = extensions->getHeap();
+	UDATA arrayletLeafSize = env->getOmrVM()->_arrayletLeafSize;
+
+	/* Adjust arraylet leaf count. There might be the case where the last leaf is NULL */
+	if (_dataSize % arrayletLeafSize == 0) {
+		arrayletLeafCount -= 1;
+	}
+
+	void *result = NULL;
+
+#define ARRAYLET_ALLOC_THRESHOLD 64
+	void *leaves[ARRAYLET_ALLOC_THRESHOLD];
+	void **arrayletLeaveAddrs = leaves;
+	if (arrayletLeafCount > ARRAYLET_ALLOC_THRESHOLD) {
+		arrayletLeaveAddrs = (void **)env->getForge()->allocate(arrayletLeafCount * sizeof(uintptr_t), MM_AllocationCategory::GC_HEAP, J9_GET_CALLSITE());
+	}
+
+	if (arrayletLeaveAddrs == NULL) {
+		return NULL;
+	}
+	UDATA elementsSize = extensions->indexableObjectModel.getDataSizeInBytes((J9IndexableObject *)objectPtr);
+
+	GC_SlotObject *slotObject = NULL;
+	uintptr_t count = 0;
 
+	while (NULL != (slotObject = arrayletLeafIterator.nextLeafPointer())) {
+		void *currentLeaf = slotObject->readReferenceFromSlot();
+		/* In some corner cases the last leaf might be NULL therefore we must ignore it */
+		if (NULL == currentLeaf) {
+			break;
+		}
+		arrayletLeaveAddrs[count] = currentLeaf;
+		count++;
+	}
+
+	GC_SlotObject objectSlot(env->getOmrVM(), &extensions->indexableObjectModel.getArrayoidPointer((J9IndexableObject *)objectPtr)[0]);
+	J9Object *firstLeafSlot = objectSlot.readReferenceFromSlot();
+
+	MM_HeapRegionDescriptorVLHGC *firstLeafRegionDescriptor = (MM_HeapRegionDescriptorVLHGC *)heap->getHeapRegionManager()->tableDescriptorForAddress(firstLeafSlot);
+
+	/* gets pagesize  or j9vmem_supported_page_sizes()[0]? */
+	UDATA pageSize = j9mmap_get_region_granularity(NULL);
+
+	/* Get heap and from there call an OMR API that will doble map everything */
+	result = heap->doubleMapArraylet(env, arrayletLeaveAddrs, count, arrayletLeafSize, elementsSize, 
+				&firstLeafRegionDescriptor->_arrayletDoublemapID,
+				pageSize);
+
+	if (arrayletLeafCount > ARRAYLET_ALLOC_THRESHOLD) {
+		env->getForge()->free((void *)arrayletLeaveAddrs);
+	}
+
+	/* Double map failed */
+	if ((NULL == result) ||	(NULL == firstLeafRegionDescriptor->_arrayletDoublemapID.address)) {
```

Comment:
`(NULL == result)` is not necessary here, is not it?

Reply:
You're correct, it's redundant. Removing it.